content/articles/how-to-build-a-random-password-generator-app-with-html-css-javascript/index.md

```diff
@@ -0,0 +1,298 @@
+
+You can access the values inside the password field by using the `.value` getter method, which returns the input value. You can also use the `.checked` getter that'll return true if the checkboxes are selected or not. It will return false if the checkboxes are not selected. 
+
+We are going to create a variable called `password`. This variable will store the value returned by the `generatePassword` function. The `generatePassword` function takes four arguments. Lastly, the `innerText` method would target the text inside `resultDOM` then change it with the generated password.
+
+Next, let's create the password generating function.
+
+```JavaScript
+// The Password Generating Function
+let generatePassword = (
+  characterAmount,
+  includeUppercase,
+  includeNumbers,
+  includeSymbols
+) => {
+  let charCodes = LOWERCASE_CODES;
+  if (includeUppercase) charCodes = charCodes.concat(UPPERCASE_CODES);
+  if (includeSymbols) charCodes = charCodes.concat(SYMBOL_CODES);
+  if (includeNumbers) charCodes = charCodes.concat(NUMBER_CODES);
+  const passwordCharacters = [];
+  for (let i = 0; i < characterAmount; i++) {
+    const characterCode =
+      charCodes[Math.floor(Math.random() * charCodes.length)];
+    passwordCharacters.push(String.fromCharCode(characterCode));
+  }
+  return passwordCharacters.join("");
+};
+```
+
+Now let me explain the codes after passing the four arguments. You want the password to be lowercase if no option is checked. Inside the function, you will create a variable that will store an array of the character codes, and you will assign the lowercase character codes by using the `let charCodes`. After that, you need to check if the options are true or not by using the conditional statements `if (includeUppercase) charCodes` for uppercase code, `if (includeSymbols) charcodes` for symbol codes, and `if (includeNumbers) charcodes` for number codes. Now depending on the options selected, you concatenate the values to the `charCodes` variable. For the password that it will create, you will create an empty array and call it `passwordCharacters`. After that, you will create a loop that will loop until it reaches the number of characters you want. While inside the loop, you will generate random character codes from the available values in the `charCodes` array. And then convert the characters from the character codes and push them into the `passwordCharacters`array so that you have completed the `generatePasswordfunction`. Now you will loop till the character amount you are getting from the input field in the app. The `charCodes` variable has all the character codes. It all depends on the options the user selected. You will now generate a random index position of the array by using the ` math.random()` method, you multiply it with the `charCodes.length` to restrict it to generate numbers up to the highest index position. Next is the `Math.floor`, which will complete the number that is generated.
```

Comment:
This paragraph is very long. Please break them into smaller paragraphs and fix the grammar. 
Refer to the edits I performed on the previous paragraphs.